src/transport/TransportMgrBase.cpp

```diff
@@ -57,6 +57,16 @@ CHIP_ERROR TransportMgrBase::MulticastGroupJoinLeave(const Transport::PeerAddres
 
 void TransportMgrBase::HandleMessageReceived(const Transport::PeerAddress & peerAddress, System::PacketBufferHandle && msg)
 {
+    if (msg->TotalLength() > CHIP_CONFIG_DEFAULT_UDP_MTU_SIZE)
```

Comment:
After careful consideration, it's very possible this will have unintended consequences on some transports and will not be future proof, especially if *more or larger headers occur, such as if there are IPv6 header options*, and will be hard to override, if this is a build-time constant. We need to let at least the Platform give this value at runtime